romancal/associations/mk_skycell_asn_from_patchlist.py

```diff
@@ -0,0 +1,149 @@
+"""Create a set of level 3 associations based on skycells"""
+
+import argparse
+import logging
+import os.path
+import sys
+
+import numpy as np
+
+import romancal.patch_match.patch_match as pm
+from romancal.associations import asn_from_list
+from romancal.lib.basic_utils import parse_visitID
+from romancal.patch_match.patch_match import get_projectioncell_wcs
+
+__all__ = ["mk_skycell_asn_from_patchlist"]
+
+# Configure logging
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+logger.setLevel("INFO")
+
+
+def mk_skycell_asn_from_patchlist(filelist, release_product, optical_element):
+    """
+    Create level 3 associations from a list of match files generated with mk_patchlist.
+
+    This function processes a list of match files, sorts them based on which input exposures
+    contribute to a given skycell and output an association with the needed exposure files.
+
+    Parameters
+    ----------
+    filelist : list or str
+        List of file names to be processed.
+    """
+
+    patch_list = []
+    for file_name in filelist:
+        with open(file_name) as match_file:
+            logger.debug(f"Working on file:{file_name}")
+            match_list = match_file.read().split(" ")
+            match_list[1] = np.fromstring(
+                match_list[1].strip("[]"), dtype="int", sep=","
+            )
+            patch_list.append(match_list)
+
+    # get a list of all patches in the match files
+    patch_array = []
+    for item in patch_list:
+        # pdb.set_trace()
+        patch_array.append(item[1])
+
+    logger.debug(f"patch_array: {patch_array}")
+    unique_patches = np.unique(np.concatenate(patch_array))
+    logger.info(f"Unique Patches: {unique_patches}")
+    pm.load_patch_table()
+    for item in unique_patches:
+        member_list = []
+        patch_name = pm.PATCH_TABLE[item]["name"]
+        for entry in patch_list:
+            if np.isin(item, entry[1]):
+                # pdb.set_trace()
+                member_list.append(os.path.basename(entry[0]).split(".")[0])
+
+        # grab all the wcs parameters needed for generate_tan_wcs
+        projcell_info = get_projectioncell_wcs(item)
+        parsed_visit_id = parse_visitID(member_list[0][1:20])
+        program_id = parsed_visit_id["Program"]
+
+        asn_file_name = (
+            "r"
+            + program_id
+            + "_"
+            + release_product
+            + "_"
+            + "v"
+            + member_list[0][1:20]
+            + "_"
+            + patch_name
+            + "_"
+            + optical_element
+        )
```

Comment:
Just looking to the future, do you think it makes sense to add a new function that skycell_asn and mk_skycell_asn_from_patchlist call that is a one-stop-shop for the components -> filename conversion?  I just worry one or the other will get out of date in the current scheme.  But that need not be part of this PR.

Reply:
Yes, this should be a function the basic_utils just like parse_visitID